nesta/core/orms/nih_orm.py

```diff
@@ -2,116 +2,122 @@
 NIH schema
 ==============
 
-The schema for the World RePORTER data. Note that the schema
-was automatically generated based on an IPython hack session
-using the limits and properties of the data from the 
-World ExPORTER (which explains the unusual VARCHAR limits).
+The schema for the World RePORTER data.
 '''
 
 from sqlalchemy.ext.declarative import declarative_base
-from sqlalchemy.dialects.mysql import VARCHAR, TEXT
-from sqlalchemy.types import INTEGER
+from sqlalchemy.dialects.mysql import VARCHAR as _VARCHAR
+from sqlalchemy.dialects.mysql import TEXT as _TEXT
+from sqlalchemy.types import INTEGER, JSON, DATETIME
 from sqlalchemy import Column, Table
 
 
 Base = declarative_base()
+TEXT = _TEXT(collation='utf8mb4_unicode_ci')
+VARCHAR = lambda *args, **kwargs: _VARCHAR(*args, 
+                                           collation='utf8mb4_unicode_ci', 
+                                           **kwargs)
+
 
 class Projects(Base):
     __tablename__ = 'nih_projects'
 
-    application_id = Column(INTEGER, primary_key=True)
+    application_id = Column(INTEGER, primary_key=True, autoincrement=False)
```

Comment:
autoincrement is one of the most annoying gotchas in sqlalchemy: it is always on by default, meaning that if you don't have a primary key present then it will assign you one. The problem with this is that when doing development you can end up with clashes, and also it doesn't "fail" if there is no PK present (which in this case there always should be).

I think just as a principle of good practice, we should always set this to False and then explicitly set to True as a way of communicating that this is an auto increment key. Seems crazy to me that this is the default behaviour!